test/PushNotificationTests/UnpackagedTests.cpp

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include <TestDef.h>
+#include "UnpackagedTests.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Microsoft::Windows::PushNotifications;
+
+
+namespace Test::PushNotifications
+{
+    HRESULT UnpackagedTests::ChannelRequestHelper(IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& channelOperation)
+    {
+        if (channelOperation.wait_for(c_timeout) != winrt::Windows::Foundation::AsyncStatus::Completed)
+        {
+            channelOperation.Cancel();
+            return HRESULT_FROM_WIN32(ERROR_TIMEOUT); // timed out or failed
+        }
+
+        auto result{ channelOperation.GetResults() };
+        auto status{ result.Status() };
+        if (status != PushNotificationChannelStatus::CompletedSuccess)
+        {
+            return result.ExtendedError(); // did not produce a channel
+        }
+
+        result.Channel().Close();
+        return S_OK;
+    }
+
+    void UnpackagedTests::ChannelRequestUsingNullRemoteId()
+    {
+        if (!m_isSelfContained)
+        {
+            VERIFY_THROWS_HR(PushNotificationManager::Default().CreateChannelAsync(winrt::guid()).get(), E_INVALIDARG);
+        }
+        else
+        {
+            auto channelOperation{ PushNotificationManager::Default().CreateChannelAsync(winrt::guid()) };
+            VERIFY_ARE_EQUAL(ChannelRequestHelper(channelOperation), E_FAIL);
+        }
+    }
+
+    void UnpackagedTests::ChannelRequestUsingRemoteId()
+    {
+        if (!m_isSelfContained)
```

Comment:
What is the purpose of the SelfContained scenario? it seems to me that for self contained almost all tests are going to fail. If so I would remove noise from this file and create tests specifically for self-contained. Also, that way you can unify a lot of tests with the Packaged ones.

Reply:
Packaged, Packaged + SelfContained, Unpackaged, all have the same test behavior. Unpackaged + SelfContained is the only scenario for PushNotifications that needs special attention since none of the APIs will fail.

Reply:
I didn't know a Packaged app could be SelfContained too, aren't they mutually exclusive?

Reply:
Nope. SelfContained just means that the binaries are shipped with the app instead of having to have a dependency on the WinAppSDK :)

Reply:
This is incorrect. We should use IsSupported(). Not m_isSelfContained